examples/ipc_file_read.rs

```diff
@@ -43,7 +43,7 @@ fn read_batch(path: &str) -> Result<(Schema, Chunk<Box<dyn Array>>)> {
         &metadata,
         None,
         chunk_index,
-        &mut vec![],
+        &mut Default::default(),
```

Comment:
would it be make sense / be possible to make the struct private? I think we could it via

```
struct ReadBuffer<'a> {
    data: &'a mut Vec<u8>,
    length: usize,
}
```

users would still use Vec in the public, and we would wrap the mut reference.

We do lose the `length` between calls, but I think that it would be less mental workload knowing about `ReadBuffer`.

Reply:
Could we make a public API that allows `Vec` as input?

We already implement `Default` and `From<Vec>`. 

So we could make an API that accepts `Vec` or `ReadBuffer` but is easily created with `Default::default()`.


For more lower level users (e.g. polars, I still want to parallize the reading) it would be easier if it is public. But we could still hide that from the default readers API ofcourse.